### ISSUE TITLE
Remove AugmentationAspect (Neos.Ui)

### ISSUE THREAD

**Issue by mhsdesign** [neos/neos-development-collection#3595]
subissue of #3216

currently f.x. `$this->contentElementWrappingService->wrapContentObject` is aop'd by the Neos.Ui to make adjustments.
Handling this via AOP should be avoided see #3216.

we could add to `Neos.Neos:ContentElementWrapping`: https://github.com/neos/neos-development-collection/blob/abd0be747f00c3b5ecae813a278044494dcb891a/Neos.Neos/Resources/Private/Fusion/Prototypes/ContentElementWrapping.fusion#L6
an api to switch the service like:

```
prototype(Neos.Neos:ContentElementWrapping) {
  @class = 'Neos\\Neos\\Fusion\\ContentElementWrappingImplementation'
  # Will be implemented f.x. via the Neos.Ui
  wrappingServiceClass = null
  node = ${node}
  value = ${value}
  # Additional attributes in the form '<attribute-name>': '<attribute-value>' that will be rendered in the ContentElementWrapping
  additionalAttributes = Neos.Fusion:DataStructure
  # This should only be set to true to render the metadata for the currently rendered document node as that is treated differently.
  renderCurrentDocumentMetadata = false
}
```

in the neos ui package then:
```
prototype(Neos.Neos:ContentElementWrapping) {
  wrappingServiceClass = 'Neos\\Neos\\Ui\\Service\\ContentElementWrappingService'
}
```

the `ContentElementEditableService` should also be refactored like this.

and maybe we can get rid of long deprecated `typoscript` attributes, which are still rendered.

### Problems

when separating this, the Neos.Ui and Neos would be required in the same version (say both at least 8.0). Currently this is more flexible i think.

**Comment by mhsdesign** [comment]
fusion alternatives:
```
prototype(Neos.Neos:ContentElementWrapping) {
    # new thing ^^
    @interface = 'Neos\\Neos\\Fusion\\ContentElementWrappingImplementationInterface'
    node = ${node}
    value = ${value}
    additionalAttributes = Neos.Fusion:DataStructure
    renderCurrentDocumentMetadata = false
}
```

> Maybe Neos.Neos:ContentElementWrapping could be an empty shell that returns ${value} and. Neos.NeosUi adds an @process to that.

```
# in Neos.Neos
prototype(Neos.Neos:ContentElementWrapping) < prototype(Neos.Fusion:Component) {
    node = ${node}
    value = ${value}
    additionalAttributes = Neos.Fusion:DataStructure
    renderCurrentDocumentMetadata = false
    renderer = ${props.value}
}
# in Neos.Neos.UI
prototype(Neos.Neos:ContentElementWrapping) {
    renderer.@process.uiWrapping {
        expression = Neos.Neos.UI:ContentElementWrapping {
            node = ${props.node}
            value = ${value}
            additionalAttributes = ${props.additionalAttributes}
            renderCurrentDocumentMetadata = ${props.renderCurrentDocumentMetadata}
        }
    }
}
```

**Comment by mhsdesign** [comment]
the easiest way for fluid and fusion would be to make a ContengWrappingServiceInterface, which will be implemented by the
 neos ui - and we ask the objectmanager to get this service...

**Comment by bwaidelich** [comment]
> the easiest way for fluid and fusion would be to make a ContengWrappingServiceInterface,
> which will be implemented by the neos ui

I agree. An Interface with implementations that can be switched via `Objects.yaml` would be better than a class name that is configured via regular Fusion value or even a new concept of `@interface`.
**But** That would mean that this behavior is _globally_ changed. Besides: I think a "ContentWrappingService" sounds rather quirky.. What's the purpose of such service. It seems to be far too generic to me.


IMO we should go one step back and ask ourselves: Why is the `\Neos\Neos\Ui\Aspects\AugmentationAspect` needed in the first place?
i.e. are there additions that might be useful for the default implementation (so we could move them there with 8.0)? And can we get rid of the other ones and/or add them via Fusion?

**Comment by mhsdesign** [comment]
I think we can keep it as is for the legacy cr but take the chance to get it right for the new cr.
 
tagging @skurfuerst here ;)

i think the Idea was to rethink the ui api in general for the new cr right? is there an open issue concerning that?
